framework/src/media/StreamBuffer.cpp

```diff
@@ -130,6 +141,49 @@ void StreamBuffer::notifyObserver(State st, ...)
 	}
 }
 
+StreamBuffer::Builder::Builder()
+	: mBufferSize(CONFIG_STREAM_BUFFER_SIZE_DEFAULT), mThreshold(CONFIG_STREAM_BUFFER_THRESHOLD_DEFAULT)
+{
+}
+
+StreamBuffer::Builder::~Builder()
+{
+}
+
+StreamBuffer::Builder &StreamBuffer::Builder::setBufferSize(size_t bufferSize)
+{
+	mBufferSize = bufferSize;
+	return *this;
+}
+
+StreamBuffer::Builder &StreamBuffer::Builder::setThreshold(size_t threshold)
+{
+	mThreshold = threshold;
+	return *this;
+}
+
+std::shared_ptr<StreamBuffer> StreamBuffer::Builder::build()
+{
+	if (mBufferSize == 0) {
+		mBufferSize = CONFIG_STREAM_BUFFER_SIZE_DEFAULT;
```

Comment:
constructor already set the CONIFG_STREAM_BUFFER_SIZE_DEFAULT

Reply:
If user do setBufferSize(0), mBufferSize can be 0

Reply:
in my opinion, each of function should check the exception case.